app/com/arpnetworking/metrics/portal/alerts/impl/DailyPartitionCreator.java

```diff
@@ -0,0 +1,283 @@
+                .addData("schema", _schema)
+                .addData("table", _table)
+                .addData("lookahead", _lookahead)
+                .log();
+    }
+
+    @Override
+    public Receive createReceive() {
+        return new ReceiveBuilder()
+                .matchEquals(START_TICKING, msg -> startTicking())
+                .matchEquals(TICK, msg -> tick())
+                .matchEquals(EXECUTE, msg -> execute())
+                .build();
+    }
+
+    private void recordCounter(final String metricName, final long value) {
+        final String fullMetric = String.format("partition_creator/%s/%s", _table, metricName);
+        _periodicMetrics.recordCounter(fullMetric, value);
+    }
+
+    // Message handlers
+
+    private void startTicking() {
+        if (getTimers().isTimerActive(TICKER_NAME)) {
+            getSender().tell(Optional.of(new IllegalStateException("Timer already started")), getSelf());
+            return;
+        }
+        LOGGER.info().setMessage("Starting execution timer")
+            .addData("schema", _schema)
+            .addData("table", _table)
+            .addData("lookahead", _lookahead)
+            .log();
+        final Optional<Exception> executeResult = execute();
+        getTimers().startPeriodicTimer(TICKER_NAME, TICK, TICK_INTERVAL);
+        getSender().tell(executeResult, getSelf());
+    }
+
+    private void tick() {
+        recordCounter("tick", 1);
+
+        final Instant now = _clock.instant();
+        if (_schedule.nextRun(_lastRun).map(run -> run.isBefore(now)).orElse(true)) {
+            getSelf().tell(EXECUTE, getSelf());
+        }
+    }
+
+    // Wrapper to propagate any errors that occurred to the caller.
+    // This is really only useful on a call to `start`.
+    private Optional<Exception> execute() {
+        final LocalDate startDate = ZonedDateTime.ofInstant(_clock.instant(), _clock.getZone()).toLocalDate();
+        final LocalDate endDate = startDate.plusDays(_lookahead);
+
+        LOGGER.info().setMessage("Creating daily partitions for table")
+                .addData("schema", _schema)
+                .addData("table", _table)
+                .addData("startDate", startDate)
+                .addData("endDate", endDate)
+                .log();
+
+        Optional<Exception> error = Optional.empty();
+        try {
+            execute(_schema, _table, startDate, endDate);
+            _lastRun = Optional.of(_clock.instant());
+        } catch (final PersistenceException e) {
+            error = Optional.of(e);
+            LOGGER.error().setMessage("Failed to create daily partitions for table")
+                    .addData("schema", _schema)
+                    .addData("table", _table)
+                    .addData("startDate", startDate)
+                    .addData("endDate", endDate)
+                    .setThrowable(e)
+                    .log();
+        }
+        recordCounter("create", error.isPresent() ? 0 : 1);
```

Comment:
This will not be executed at all if there is a runtime exception.

Reply:
meant to put this in a finally 😄